pkg/server/handler/v1alpha1/integration.go

```diff
@@ -349,43 +339,59 @@ func getIntegration(tenant, name string) (*api.Integration, error) {
 	return SecretToIntegration(secret)
 }
 
-// UpdateIntegration updates an integration with the given tenant name and integration name.
-// If updated successfully, return the updated integration.
-func UpdateIntegration(ctx context.Context, tenant, name string, in *api.Integration) (*api.Integration, error) {
-	if in.Spec.Type == api.Cluster && in.Spec.Cluster != nil {
-		oldIn, err := getIntegration(tenant, name)
+func updateClusterIntegration(tenant, name string, in *api.Integration) error {
+	oldIn, err := getIntegration(tenant, name)
+	if err != nil {
+		log.Errorf("get integration %s error %v", name, err)
+		return err
+	}
+
+	oldCluster := oldIn.Spec.Cluster
+	cluster := in.Spec.Cluster
+	if cluster.Namespace == "" {
+		cluster.Namespace = oldCluster.Namespace
+	} else if cluster.Namespace == oldCluster.Namespace && cluster.PVC == "" {
+		cluster.PVC = oldCluster.PVC
+	}
+	// turn on worker cluster
+	if !oldCluster.IsWorkerCluster && cluster.IsWorkerCluster {
```

Comment:
Handle these four cases by two-layer nesting if-else?

Reply:
I want to keep the current form since it is easy to understand.